### DISCUSSION TITLE
Twig two columns, different content.

### DISCUSSION THREAD

**Discussion by vmdevmihajlo** [sulu/sulu#7032] [Q&A] [answered]
Hello, i need little help. I got stuck with this, i have two columns and i want to allow users of cms choose blocks they want. i know this is not good but i dont know where to find this solution. This just duplicate my blocks into both columns...
![Screenshot_20230308_123706](https://user-images.githubusercontent.com/98061506/223703621-666c9791-3cd0-4b1e-8a07-df137d4bfab5.png)

**Comment by alexander-schranz** [comment, marked as answer]
You can just extend your XML to have `right-blocks` and `left-blocks`, but mostly that is something I would not recommend as you should always not bind your Template to deep to your design.


The best advise I can give you in structuring your `XML` templates is, to go a step back and ignore first all kind of things which has todo how the content is displayed – Topic ["separating content from presentation"](https://sulu.io/guides/templating-and-structured-content-in-sulu). Your template at first should not care if it is 2 column, Red, Blue whatever. First structure the Content and your Blocks the way how a ScreenReader would read your content, the mobile design were mostly also not such complex things exist is also a good start point. But I really would go that wide back by looking at the content structure first how a screen reader user would read your page.

Aftering so inserting all your content in some kind unrelated to your design, just how Screen Readers also Crawlers read your page. Most of the blocks should be able to be displayed they way you already want it with CSS. For some blocks you maybe want to add some more checkboxes, selects, to give the Content Manager possibility to change how things are outputted.

But in my opinion Template and Blocks should be created that way that you can do every year without a problem a redesign that and don't need to create migrations for the XML templates only for some presentation fields but real content (text, images) should stay as it is for the future.

But that is just my advice. If you want to create some kind of layout via `two-column` block or so on you can still do it Sulu allows it. It just something I always want to mentino that structuring your XML to deep connected to the design will make redesigns not so easy. So I'm more the guy to say structure your content how a screen reader and crawler see it and then extend it with some few fields for presentation but keep then as small as possible.
